### PR TITLE
[WIP] Frontend microcomponents

### PR DESCRIPTION
**Pull request by victorporof**
This is a WIP that I'd like to land just so it doesn't get bitrotted. I specifically avoided locationbar and navbar changes for now. Most important things is having a widgets directory and shared prop types.

### CODE REVIEW

**Comment by victorporof** [comment]
Relevant changes start with 30ae952.

**Comment by victorporof** [comment]
@jsantell

File: app/ui/browser/views/browser-chrome.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+const BrowserChrome = function(props) {
+  return (
+    <div className={CHROME_AREA_STYLE}>
+      <NavBar {...props} />
+      <TabBar {...props} />
+    </div>
+  );
+};
+
+BrowserChrome.displayName = 'BrowserChrome';
```
Comment (jsantell) [inline app/ui/browser/views/browser-chrome.jsx]: We should probably avoid the usage of `Chrome` in describing the UI, unfortunately

File: app/ui/browser/widgets/btn.jsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
+/*
```
Comment (jsantell) [inline app/ui/browser/widgets/btn.jsx]: Should probably git move this file

File: build/const.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import path from 'path';
```
Comment (jsantell) [inline build/const.js]: s/const/constants to be consistent with other similar files, and remove ambiguity of JS const

File: package.json [hunk 4]
```diff
@@ -34,15 +35,18 @@
...
     "redux": "3.5.1",
     "redux-immutable": "3.0.6",
     "redux-logger": "2.6.1",
     "request": "2.71.0",
+    "source-map-support": "ncalexan/node-source-map-support#fileUrls-plus",
     "sqlite3": "3.1.3",
     "thenify": "3.2.0",
     "thenify-all": "1.6.0",
     "true-html-escape": "0.0.2",
+    "uuid": "2.0.2",
```
Comment (jsantell) [inline package.json]: oops

File: package.json [hunk 5]
```diff
@@ -77,12 +80,11 @@
     "mocha-eslint": "2.0.2",
     "nyc": "6.4.0",
     "react-addons-perf": "15.0.1",
+    "require-dir": "^0.3.0",
```
Comment (jsantell) [inline package.json]: Use explicit versions

File: test/ui/browser/widgets/test-btn.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
+// Any copyright is dedicated to the Public Domain.
```
Comment (jsantell) [inline test/ui/browser/widgets/test-btn.js]: git move

File: app/ui/browser/widgets/btn.jsx (hunk 2 above)
Comment (victorporof) [inline app/ui/browser/widgets/btn.jsx]: I did.

**Pull request closed without merging**